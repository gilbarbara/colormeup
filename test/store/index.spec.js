import { store, persistor } from 'app-store';

jest.mock('modules/helpers', () => {
  const helpers = require.requireActual('modules/helpers');

  return {
    ...helpers,
    getUnixtime: () => 1234567890,
  };
});

describe('store', () => {
  it('should have a store', () => {
    expect(store.getState()).toEqual({
      _persist: { rehydrated: true, version: -1 },
      app: { alerts: [], isSidebarActive: false, version: 3 },
      color: { hex: '', instance: null, model: 'hsl', steps: 24, type: 'h' },
      router: { location: null },
      user: {
        colors: [],
        createdAt: 1234567890,
        showHelp: true,
        showStarterKit: true,
        updatedAt: 1234567890,
      },
    });
  });

  it('should have a persistor', () => {
    expect(persistor.getState()).toEqual({
      bootstrapped: true,
      registry: [],
    });
  });
});
