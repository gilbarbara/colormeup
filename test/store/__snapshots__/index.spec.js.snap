// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store should have a store 1`] = `
Object {
  "_persist": Object {
    "rehydrated": true,
    "version": -1,
  },
  "app": Object {
    "alerts": Array [],
    "isSidebarActive": false,
    "version": 3,
  },
  "color": Object {
    "hex": "",
    "instance": null,
    "model": "hsl",
    "steps": 24,
    "type": "h",
  },
  "router": Object {
    "action": "POP",
    "location": Object {
      "hash": "",
      "pathname": "/",
      "query": Object {},
      "search": "",
      "state": Object {},
    },
  },
  "user": Object {
    "colors": Array [],
    "createdAt": 1234567890,
    "showHelp": true,
    "showStarterKit": true,
    "updatedAt": 1234567890,
  },
}
`;
